src/System.Net.Sockets/tests/FunctionalTests/TcpClientTest.cs

```diff
@@ -36,16 +39,32 @@ public TcpClientTest(ITestOutputHelper output)
                 string host = System.Net.Test.Common.Configuration.Sockets.SocketServer.IdnHost;
                 int port = System.Net.Test.Common.Configuration.Sockets.SocketServer.Port;
 
-                if (mode == 0)
-                {
-                    await client.ConnectAsync(host, port);
-                }
-                else
+                IPAddress[] addresses;
+                switch (mode)
                 {
-                    IPAddress[] addresses = await Dns.GetHostAddressesAsync(host);
-                    await (mode == 1 ?
-                        client.ConnectAsync(addresses[0], port) :
-                        client.ConnectAsync(addresses, port));
+                    case 0:
+                        await client.ConnectAsync(host, port);
+                        break;
+                    case 1:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await client.ConnectAsync(addresses[0], port);
+                        break;
+                    case 2:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await client.ConnectAsync(addresses, port);
+                        break;
+
+                    case 3:
+                        await Task.Factory.FromAsync(client.BeginConnect, client.EndConnect, host, port, null);
+                        break;
+                    case 4:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await Task.Factory.FromAsync(client.BeginConnect, client.EndConnect, addresses[0], port, null);
+                        break;
+                    case 5:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await Task.Factory.FromAsync(client.BeginConnect, client.EndConnect, addresses, port, null);
+                        break;
```

Comment:
Thanks, but I'm not seeing why it's better.  Each case requires another 8 lines that just obscures the differences between all of the test cases.

Would it ease your concerns if I added an enum to use instead of numbers 0 through 5?